manager/dedup.go

```diff
@@ -342,12 +353,23 @@ START:
 		allowStale = true
 	}
 
-	// Stop listening if we're stopped
-	select {
-	case <-d.stopCh:
-		return
-	default:
+	if meta.LastIndex == lastIndex {
+		log.Printf("[TRACE] (dedup) %s no new data (index was the same)", path)
+		goto START
+	}
+
+	if meta.LastIndex < lastIndex {
+		log.Printf("[TRACE] (dedup) %s had a lower index, resetting", path)
+		lastIndex = 0
+		goto START
+	}
+	lastIndex = meta.LastIndex
+
+	if reflect.DeepEqual(pair, lastPair) {
```

Comment:
I needed that when I wrote the poller on the original Consul Template. I found the index could change if someone updated the key to the same value.